src/ng/directive/ngClass.js

```diff
@@ -8,92 +8,93 @@
 
 function classDirective(name, selector) {
   name = 'ngClass' + name;
-  return ['$animate', function($animate) {
+  return ['$animate','$parse', function($animate,$parse) {
     return {
       restrict: 'AC',
-      link: function(scope, element, attr) {
-        var oldVal;
-
-        scope.$watch(attr[name], ngClassWatchAction, true);
-
-        attr.$observe('class', function(value) {
-          ngClassWatchAction(scope.$eval(attr[name]));
+      compile: function(tElement, tAttr) {
+        var classFn = $parse(tAttr[name], function(value) {
```

Comment:
I'll put it again inside a link function. 
I suggest that it can be considered a "major upgrade" to do such refactor of all parse expression code from link functions to compile functions, in order to improve performance when ng-repeat & others are involved.